src/SFA.DAS.CaptureTheFlag.Web/Services/ForumService.cs

```diff
@@ -0,0 +1,75 @@
+﻿using DAS_Capture_The_Flag.Models.Forum;
+using Microsoft.AspNetCore.Identity;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using DAS_Capture_The_Flag.Data;
+using Microsoft.EntityFrameworkCore;
+
+namespace DAS_Capture_The_Flag.Services
+{
+    public class ForumService : IForumService
+    {
+        // Pass an instant of our dbContext
+        private readonly ApplicationDbContext _db;
+
+        public ForumService(ApplicationDbContext db)
+        {
+            _db = db;
+        }
+
+        public void Create(Forum forum)
+        {
+            _db.Forums.Add(forum);
+            _db.SaveChanges();
+        }
+
+        public void Delete(int forumId)
+        {
+            var topic = _db.Forums.Find(forumId);
+            _db.Forums.Remove(topic);
+        }
+
+        public IEnumerable<Forum> GetAll() // returns instance of forum from our database
+        {
+            return _db.Forums.Include(f => f.Posts);
```

Comment:
Future Improvement - Whilst enumerating the forums for performance reasons I probably wouldn't get the posts at the same time as there could be a lot of posts n a forum.

There could also be a lot of forums if the site was really popular so paging the forums would be a good idea
See the .Skip and .Take methods on the forum collection